library/extractor/src/main/java/com/google/android/exoplayer2/extractor/mp4/FragmentedMp4Extractor.java

```diff
@@ -968,6 +970,27 @@ private static long parseTfdt(ParsableByteArray tfdt) {
     return version == 1 ? tfdt.readUnsignedLongToLong() : tfdt.readUnsignedInt();
   }
 
+  /**
+   * Retunrs if edts list duration is for the entire media timeline.
+   *
+   * @param track The {@link Track} for which edts list duration to be checked.
+   * @return flag indicating if the edts list duration is for entire timeline.
+   */
+  private static boolean isEdtsListDurationForEntireMediaTimeline(Track track) {
+    // Currently we only support a single edit that moves the entire media timeline (indicated by
+    // duration == 0 or duration == track duration).
+    // Other uses of edit lists are uncommon and unsupported.
+    if (track.editListDurations != null && track.editListDurations.length == 1) {
+      long edtsDurationUs = Util.scaleLargeTimestamp(track.editListDurations[0],
+          C.MICROS_PER_SECOND, track.movieTimescale);
+      return edtsDurationUs == 0
+          || (track.durationUs > MICROS_PER_MILLI
+          && edtsDurationUs >= track.durationUs - MICROS_PER_MILLI
+          && edtsDurationUs <= track.durationUs + MICROS_PER_MILLI);
```

Comment:
The `edtsListDuration` is the duration of the edit itself. So the end time of the edit is `editListMediaTime + editListDuration`. And I think if we want to check if the edit list covers the entire segment, we can accept all edits that exceed the track duration. So the condition can just be:
`scaleToUs(editListMediaTime + editListDuration) >= track.durationUs`

Reply:
Agreed. Updated the code @tonihei .